### DISCUSSION TITLE
Adding realtime suggestions and hashtag Creation to: Textarea with Inline Combobox

### DISCUSSION THREAD

**Discussion by gielcobben** [ariakit/ariakit#2225] [Questions] [answered]
Hello there 👋! I am using the Textarea with inline Combobox and I would like to add a feature that updates the suggested list in real-time as the user types, using fetch to retrieve the data. Could you please guide me on where to add this functionality in the code?

Also, it would be useful if the popover displayed an option to create a new hashtag when a search for a hashtag does not yield any matching results. For example, it could say "Create #design" or "Create #code" to help the user easily create a new hashtag.

**Comment by diegohaz** [comment, marked as answer]
Hello!

> I am using the Textarea with inline Combobox and I would like to add a feature that updates the suggested list in real-time as the user types, using fetch to retrieve the data.

Combobox has an internal algorithm to filter the list of suggestions based on the `combobox.list` and `combobox.value` properties of the combobox state. The result is stored in the `combobox.matches` property. But you don't have to use those properties. They are there for convenience. At its core, Combobox is only concerned with the `ComboboxItem` elements that are rendered with a `value` prop. If you want to filter the list yourself, you can ignore `list` and `matches` and render your own list of `ComboboxItem`s based on a state you maintain and populate with fetch (or any other means).

> Also, it would be useful if the popover displayed an option to create a new hashtag when a search for a hashtag does not yield any matching results. For example, it could say "Create #design" or "Create #code" to help the user easily create a new hashtag.

Just render a new `ComboboxItem` with no `value` prop at the end of the list:

```jsx
{myOwnResultsBasedOnComboboxDotValue.map((result) => (
  <ComboboxItem ...
))}
<ComboboxItem onClick={createHashtag}>Create #{combobox.value}</ComboboxItem>
```
